### PR TITLE
Fix EZP-24315: Add support for sorting in REST Views

### PR DESCRIPTION
**Pull request by glye**
> Implements https://jira.ez.no/browse/EZP-24315
> Status: Ready for review

Continues on the work from @Nattfarinn in https://github.com/ezsystems/ezpublish-kernel/pull/1651 - Thank you!

- [x] Renamed `GenericDataKey` class to `DataKeyValueObjectClass`
- [x] Added missing docblock param types
- [x] Fixed exception message parameter
- [x] Check if the data key exists before using it, throw if it doesn't
- [x] Docblock description for `$dataKey` and `$valueObjectClass`
- [x] Unit test for the DataKey parser
- [x] Update REST spec

### CODE REVIEW

**Comment by glye** [comment]
Review ping @Nattfarinn @bdunogier @andrerom

File: doc/specifications/rest/REST-API-V2.rst [hunk 1]
```diff
@@ -2173,9 +2173,7 @@ Perform a query on images withing the media section, sorted by name, limiting re
         <limit>10</limit>
         <offset>0</offset>
         <SortClauses>
-          <SortClause>
-            <SortField>NAME</SortField>
-          </SortClause>
+          <ContentName>ascending</ContentName>
```
Comment (glye) [inline doc/specifications/rest/REST-API-V2.rst]: @Nattfarinn This is how I understood it, anyway. Please verify.

**Comment by bdunogier** [comment]
Looks okay so far. But shouldn't this also include the other sort clauses that are supported ? Unless I am mistaken, only DatePublished is actually defined.

**Comment by glye** [comment]
@bdunogier Right, added.

**Comment by bdunogier** [comment]
Were you able to test them all ? I'm wondering if Field is gonna be that simple... :-)

**Comment by andrerom** [comment]
> Were you able to test them all ? I'm wondering if Field is gonna be that simple... :-)

indeed not, will need to have an attribute for it in this form.

**Comment by glye** [comment]
Ack. `MapLocation` will need this too, it seems.

**Comment by glye** [comment]
@andrerom @bdunogier Removed `Field` and `MapLocationDistance`, postponing them due to complexity, and they seem not fully implemented: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/REST/Server/Input/Parser/Criterion/Field.php#L33

All others verified in manual testing with chrome rest client.

**Comment by andrerom** [comment]
ok with me, ok with you to @bdunogier ?

**Comment by bdunogier** [comment]
+1 then.

**Comment by bdunogier** [comment]
> ok with me for now (6.3), ok with you to @bdunogier ?

Hmm, 6.3 ? It's a new feature, and nothing can depend on it Any particular reason why we'd want it before 6.4 ?

**Comment by andrerom** [comment]
> Hmm, 6.3 ? It's a new feature, and nothing can depend on it Any particular reason why we'd want it before 6.4 ?

sorry, meant 6.4, implying we get this in for 1.4, and take fields for 1.5 :)

**Comment by bdunogier** [comment]
> sorry, meant 6.4, implying we get this in for 1.4, and take fields for 1.5 :)

😌 

He we go then.

**Pull request merged**